### ISSUE TITLE
Unfolding a region unfolds sub-regions

### ISSUE THREAD

**Issue by ipsod** [syl20bnr/spacemacs#5480]
In Vim, folding a region does not change the fold-state of subregions.  In Spacemacs, unfolding a region unfolds all sub-regions.

To reproduce, create a new buffer, activate python-mode, and paste this text:
```
class Test():
    def whatever():
        test = 'test'
        test = 'test'
```

Fold the method, and then fold the class, then unfold the class.  The method should have unfolded as well.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.14
- Spacemacs branch: develop (rev. d52eb41)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git markdown org spell-checking unimpaired python html php django version-control auto-completion syntax-checking evil-snipe search-engine gnus
            (shell :variables shell-default-height 30 shell-default-position 'bottom))
```

**Comment by TheBB** [comment]
Evil doesn't implement its own folding operations, but falls back on existing packages. This is very likely to be considered a wontfix by them and also by us.

**Comment by TheBB** [comment]
What *exact* keybindings are you using? I can't reproduce your problem.

**Comment by TheBB** [comment]
You can try installing vimish-fold, which is a folding backend that presumably works more closely to vim behaviour.

**Comment by ipsod** [comment]
I checked out `vimish-fold`, but I think it is more for manual folding.  I couldn't find any way to do syntax-based folding with it.

**Comment by nixmaniack** [comment]
I had briefly tried [origami.el](https://github.com/gregsexton/origami.el) and it works as you described, i.e. opening class fold doesn't open subfolds. A point to be noted here is that it doesn't have vim bindings and you might even would need to write simple parsers if needed. I merely used `origami-close-node` and `origami-open-node` on supported modes.

**Comment by TheBB** [comment]
Now that we have origami in develop, this is fixed.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
